system/system_windows.go

```diff
@@ -69,6 +70,45 @@ func removeUninstallEntry(name string) error {
 	return registry.DeleteKey(registry.LOCAL_MACHINE, reg)
 }
 
+func uninstallString(installSource, extension string) string {
+	productroot := `SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\`
+	k, err := registry.OpenKey(registry.LOCAL_MACHINE, productroot, registry.ENUMERATE_SUB_KEYS)
+	if err != nil {
+		return ""
+	}
+	reg, _ := k.ReadSubKeyNames(-1)
+	err = k.Close()
+	for _, v := range reg {
+		q, err := registry.OpenKey(registry.LOCAL_MACHINE, fmt.Sprintf("%s%s", productroot, v), registry.ALL_ACCESS)
+		if err != nil {
+			continue
+		}
+		switch extension {
+		case "msi":
+			a, _, err := q.GetStringValue("InstallSource")
+			if err != nil {
+				// InstallSource not found, move on to next entry
+				continue
+			}
+			if strings.Contains(a, installSource) {
+				un, _, err := q.GetStringValue("UninstallString")
+				if err != nil {
+					// UninstallString not found, move on to next entry
+					continue
+				}
+				return un
+			}
+		}
+
+		q.Close()
+	}
+
+	if err != nil {
```

Comment:
This should likely be logged instead of using fmt.Println. Put it inside the defer above.